### PR TITLE
Adding toHaveStyle Matcher

### LINKED ISSUE #203: Implement toHaveStyle matcher
Example
```
expect(label).toHaveStyle({
    "font-family": "Faktum",
    "font-size": "26px",
    color: "#000" // handles "black", "#000000", "rgb(0, 0, 0)", and "rgba(0, 0, 0, 1)"
});
```

according to https://github.com/webdriverio/expect-webdriverio/issues/105

Please also add:
- types
- docs
- unit tests

### PR DESCRIPTION
**Pull request by Harry-Liversedge**
* Adds a toHaveStyle matcher as set out in #203 
* Adds tests for the toHaveStyle matcher in the same style as other matchers

closes #203

### CODE REVIEW

**Review by christian-bromann — APPROVED** [review]
👍  awesome, thanks so much

File: src/utils.ts [hunk 1]
```diff
@@ -146,6 +147,35 @@ export const compareTextWithArray = (actual: string, expectedArray: Array<string
     }
 }
 
+export const compareStyle = async (actualEl: WebdriverIO.Element, style: { [key: string]: string; }, { ignoreCase = true, trim = false }) => {
+    let result: Boolean = true
+    let actual: any = {}
```
Comment (christian-bromann) [inline src/utils.ts:152]:
```suggestion
    const actual: any = {}
```

File: src/utils.ts [hunk 2]
```diff
@@ -146,6 +147,35 @@ export const compareTextWithArray = (actual: string, expectedArray: Array<string
     }
 }
 
+export const compareStyle = async (actualEl: WebdriverIO.Element, style: { [key: string]: string; }, { ignoreCase = true, trim = false }) => {
+    let result: Boolean = true
```
Comment (christian-bromann) [inline src/utils.ts:151]:
```suggestion
    let result = true
```

File: src/utils.ts [hunk 3]
```diff
@@ -146,6 +147,35 @@ export const compareTextWithArray = (actual: string, expectedArray: Array<string
...
+    let actual: any = {}
+
+    for (let key in style) {
```
Comment (christian-bromann) [inline src/utils.ts:154]:
```suggestion
    for (const key in style) {
```

**Comment by christian-bromann** [comment]
Can you add the new matcher to this list as well: https://github.com/webdriverio/expect-webdriverio/blob/main/test/matchers.test.ts ?

**Comment by Harry-Liversedge** [comment]
> Can you add the new matcher to this list as well: https://github.com/webdriverio/expect-webdriverio/blob/main/test/matchers.test.ts ?

Yes I've made that change, thanks for fixing the syntax in utils!

**Comment by christian-bromann** [comment]
Thanks again, will be part of the next v3.0.0 release.

**Comment by Harry-Liversedge** [comment]
Awesome thanks so much!

**Pull request merged**